### DISCUSSION TITLE
Why are not `jacob0` and `pin.WORLD` consistent for computing jacobian wrt the world frame?

### DISCUSSION THREAD

**Discussion by LHT-Liu** [stack-of-tasks/pinocchio#2422] [Q&A] [answered]
Hi!

**My setup**
Chip: Apple M1 Pro 
macOS: Sequoia 15.0
PyCharm 2024.2.1 (Community Edition)
matlab: R2021b
pinocchio: 3.2.0

**My question:**  I use Pinocchio and  [[robotics toolbox 10.4](https://petercorke.com/toolboxes/robotics-toolbox/)](https://petercorke.com/toolboxes/robotics-toolbox/)  of matlab for my research on forward kinematics and jacobian matrix of the same position of Franka. And i found that the forward kinematics were consistent, but the function `jacobe` relative to the local coordinate system in matlab is the same as `pin.LOCAL` in pinocchio. The  function `jacob0` in matlab relative to the world coordinate system is the same as `pin.LOCAL WORLD ALIGNED` in Pinocchio. 

Why are not `jacob0` and `pin.WORLD` consistent?

what’s going on?

**In Pinocchio**
```
Flange_pose = data.oMf[Flange_Id]
print("Flange_pose \n", Flange_pose)

Flange_Jaco_wrt_local = pin.computeFrameJacobian(model, data, q, Flange_Id, pin.LOCAL)
print("Flange_Jaco_wrt_local \n", Flange_Jaco_wrt_local)


Flange_Jaco_wrt_world = pin.computeFrameJacobian(model, data, q, Flange_Id, pin.LOCAL_WORLD_ALIGNED)
print("Flange_Jaco_wrt_world \n", Flange_Jaco_wrt_world)
```
<img width="694" alt="image" src="https://github.com/user-attachments/assets/fa148819-2868-42a8-b785-eb9784281fda">

**In matlab**
```
mdl_panda

panda.tool.t = [0;0;0];

q = [0.1,0.2,0.3,0.4,0.5,0.6,0.7];

panda.fkine(q)

panda.jacob0(q)
panda.jacobe(q)
```
<img width="595" alt="image" src="https://github.com/user-attachments/assets/4fadbb30-81c1-4c40-b795-fd5e5c4762f8">

Thanks

@jviereck @jcarpent

**Comment by jcarpent** [comment, marked as answer]
> Why are not `jacob0` and `pin.WORLD` consistent?

pin.WORLD represent the point coincident with the origin of the WORLD as moving with the body.
pin.LOCAL_WORLD_ALIGNED is centered on the point of the moving body in the coordinate systems with axes aligned with the WORLD frame.

**Reply by LHT-Liu** [reply]
thanks for your reply. but i don't think you've solved my question. 
@jcarpent 
don't give me the definition.

the result of the jacob0 of the Matlab . The results of matlab's `jcaob0` function are actually consistent with the results of pinocchio's `pin.LOCAL_WORLD_ALIGNED`. Why aren't the results of matlab's` jcaob0` function consistent with the results of pinocchio's `pin.WORLD`?

<img width="655" alt="image" src="https://github.com/user-attachments/assets/a462db75-beac-4041-808e-970a08512ef8">

<img width="1362" alt="image" src="https://github.com/user-attachments/assets/885cda1a-d3d0-4f9e-9512-cf52a04e1cf2">


<img width="1167" alt="image" src="https://github.com/user-attachments/assets/dc616db2-2078-461c-94ab-7c4e180bab78">


<img width="1156" alt="image" src="https://github.com/user-attachments/assets/eeb3ed07-5e1c-4b17-bc1e-acd185b9b926">

thanks

**Reply by LHT-Liu** [reply]
@jcarpent , hello, can you help me solve my question? Thanks.
